woocommerce/woocommerce-android
# Shipping rates: rate selection

Closes: #12285 

### Description
This PR adds shipping rates items UI. With the changes included here, it is now possible to select (expand) shipping rates and select/unselect the signature-required options. This PR also updates the RoundedCornerBoxWithBorder composable, allowing it to receive border and background parameters. This PR focuses only on UI; the components here will be connected with real data in a different PR.

### Testing information

1. Create an Order with the processing status on a store that can create shipping labels
2. Tap on Orders
3. Tap on the order created on step 1
4. Tap con create shipping labels
5. Check that the shipping service card is visible
6. Tap on different carriers and check that the card information changes
7. Tap on a shipping rate and check that when selected, the shipping rate information expands
8. Tap on a signature required option and check that it selects
9. Tap on the selected option for the required signature and ensure that it deselects.

### The tests that have been performed
1. Test that the UI interaction works as expected

### Images/gif

https://github.com/user-attachments/assets/355e7dd1-0fe3-4e65-b337-e3731911dd9a


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
